packages/web/src/components/data-entry/AccessAndSaleModalLegacy.tsx

```diff
@@ -0,0 +1,190 @@
+    const isCollectibleGated =
+      isPremiumContentCollectibleGated(premiumConditions)
+    const initialValues = {}
+    set(initialValues, IS_UNLISTED, isUnlisted)
+    set(initialValues, IS_PREMIUM, isPremium)
+    set(initialValues, PREMIUM_CONDITIONS, premiumConditions)
+
+    let availabilityType = TrackAvailabilityType.PUBLIC
+    if (isUsdcGated) {
+      availabilityType = TrackAvailabilityType.USDC_PURCHASE
+    }
+    if (isFollowGated || isTipGated) {
+      availabilityType = TrackAvailabilityType.SPECIAL_ACCESS
+    }
+    if (isCollectibleGated) {
+      availabilityType = TrackAvailabilityType.COLLECTIBLE_GATED
+    }
+    if (isUnlisted) {
+      availabilityType = TrackAvailabilityType.HIDDEN
+    }
+    set(initialValues, AVAILABILITY_TYPE, availabilityType)
+    set(initialValues, FIELD_VISIBILITY, fieldVisibility)
+    set(initialValues, PREVIEW, preview)
+    set(
+      initialValues,
+      SPECIAL_ACCESS_TYPE,
+      isTipGated ? SpecialAccessType.TIP : SpecialAccessType.FOLLOW
+    )
+    return initialValues as AccessAndSaleFormValues
+  }, [fieldVisibility, isPremium, isUnlisted, premiumConditions, preview])
+
+  const onSubmit = (values: AccessAndSaleFormValues) => {
+    let newState = {
+      ...metadataState,
+      is_premium: !isEmpty(premiumConditions),
+      premium_conditions: values[PREMIUM_CONDITIONS],
+      unlisted: values.is_unlisted,
+      preview_start_seconds: values[PREVIEW] ?? 0
+    }
+
+    if (
+      get(values, AVAILABILITY_TYPE) === TrackAvailabilityType.USDC_PURCHASE
+    ) {
+      // @ts-ignore the field has a string in it
+      const priceStr: string = (
+        values[PREMIUM_CONDITIONS] as PremiumConditionsUSDCPurchase
+      ).usdc_purchase.price
+      const price = priceStr ? parseFloat(priceStr) * 100 : 0 // TODO: better default?
+      newState.premium_conditions = {
+        // @ts-ignore splits get added in saga
+        usdc_purchase: {
+          price
+        }
+      }
+    }
+
+    if (get(values, AVAILABILITY_TYPE) === TrackAvailabilityType.HIDDEN) {
+      newState = {
+        ...newState,
+        ...(get(values, FIELD_VISIBILITY) ?? undefined),
+        unlisted: true
+      }
+    } else {
+      newState = {
+        ...newState,
+        ...defaultFieldVisibility,
+        unlisted: false
+      }
+    }
+
+    didUpdateState(newState)
+  }
+
+  let availabilityButtonTitle = messages.public
+  let AvailabilityIcon = IconVisibilityPublic
+  if (isUnlisted) {
+    availabilityButtonTitle = messages.hidden
+    AvailabilityIcon = IconHidden
+  } else if (isPremium) {
+    if (premiumConditions && 'nft_collection' in premiumConditions) {
+      availabilityButtonTitle = messages.collectibleGated
+      AvailabilityIcon = IconCollectible
+    } else {
+      availabilityButtonTitle = messages.specialAccess
+      AvailabilityIcon = IconSpecialAccess
+    }
+  }
+
+  return (
+    <ContextualMenu
```

Comment:
I can take this since I was the one to add these changes, lol Andrew called them modalform so much more in line with what you're saying. I'm basically following the naming conventions in the Figma

Reply:
Yeah I have to say Contextual Menu doesn't really resonate with me, but I'm all for consistency between design and code components. Maybe worth a quick sync with design about how they conceptualize this pattern and align on a name?

Reply:
Yeah down with that. I just want the the name to be the same for mobile and web so it's clear for folks